### PR TITLE
implements pretty printing

### PR DESCRIPTION
**Pull request by KFlash**

### CODE REVIEW

File: src/printer2/index.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    source,
+    containerPos: 0,
+    containerEnd: 0,
+    declarationListContainerEnd: 0,
+    lineMap: [],
+    detachedCommentsInfo: undefined,
+  };
+}
+
+export function printSource(root: RootNode): any {
```
Comment (KFlash) [inline src/printer2/index.ts:24]: @aladdin-add What do you think? Too expensive? I had to build a container around the 'RootNode' and the same around statements, expressions and types...

**Comment by KFlash** [comment]
@aladdin-add Can you check this code? This is basically how it works. Everything is wrapped inside an container to keep track of the dangling comments. The loop invoke the container code and the comment get attached.

**Comment by KFlash** [comment]
@aladdin-add Is there a place we can chat?

**Comment by aladdin-add** [comment]
let's make the private chat here: https://discord.com/channels/857401897564372992/857402342051020820

**Comment by KFlash** [comment]
https://discord.gg/UPdYekdE

**Comment by KFlash** [comment]
@aladdin-add I will land this within a few days. As of now it finally print valid JS code and also the types get printed correctly. Comments is 98% done. Left now is to fix performance and clean up the code base.

This will be a huge PR and change it all!

**Review by aladdin-add — APPROVED** [review]

**Comment by KFlash** [comment]
@aladdin-add This is soon to land, but any ideas on how to handle the comment formatting without use of bitwise masks?

**Comment by KFlash** [comment]
@aladdin-add Can you try something?

**Comment by KFlash** [comment]
@aladdin-add Can you solve the conflicts with a force push?

**Comment by KFlash** [comment]
Closing and pushing to main

**Pull request closed without merging**